src/wizard/wizard-directive.js

```diff
@@ -0,0 +1,677 @@
+        var enabledSteps = $scope.getEnabledSteps();
+
+        // Save the step  you were on when next() was invoked
+        var index = stepIdx($scope.selectedStep);
+
+        if ($scope.selectedStep.substeps) {
+          if ($scope.selectedStep.next(callback)) {
+            return;
+          }
+        }
+
+        // Check if callback is a function
+        if (angular.isFunction(callback)) {
+          if (callback($scope.selectedStep)) {
+            if (index === enabledSteps.length - 1) {
+              this.finish();
+            } else {
+              // Go to the next step
+              if (enabledSteps[index + 1].substeps) {
+                enabledSteps[index + 1].resetNav();
+              }
+            }
+          } else {
+            return;
+          }
+        }
+
+        // Completed property set on scope which is used to add class/remove class from progress bar
+        $scope.selectedStep.completed = true;
+
+        // Check to see if this is the last step.  If it is next behaves the same as finish()
+        if (index === enabledSteps.length - 1) {
+          this.finish();
+        } else {
+          // Go to the next step
+          $scope.goTo(enabledSteps[index + 1]);
+        }
+      };
+
+      this.previous = function (callback) {
+        var index = stepIdx($scope.selectedStep);
+
+        if ($scope.selectedStep.substeps) {
+          if ($scope.selectedStep.previous(callback)) {
+            return;
+          }
+        }
+
+        // Check if callback is a function
+        if (angular.isFunction(callback)) {
+          if (callback($scope.selectedStep)) {
+            if (index === 0) {
+              throw new Error("Can't go back. It's already in step 0");
+            } else {
+              $scope.goTo($scope.getEnabledSteps()[index - 1]);
+            }
+          }
+        }
+      };
+
+      this.finish = function () {
+        if ($scope.onFinish) {
+          if ($scope.onFinish() !== false) {
+            this.reset();
+          }
+        }
+      };
+
+      this.cancel = function () {
+        if ($scope.onCancel) {
+          if ($scope.onCancel() !== false) {
+            this.reset();
+          }
+        }
+      };
+
+      //reset
+      this.reset = function () {
+        //traverse steps array and set each "completed" property to false
+        angular.forEach($scope.getEnabledSteps(), function (step) {
+          step.completed = false;
+        });
+        //go to first step
+        this.goTo(0);
+      };
+    },
+    link: function ($scope, $element, $attrs) {
+      // watching isOpen attribute to display modal when needed
+      var isOpenListener = $scope.$watch('isOpen', function (newVal, oldVal) {
+        if (newVal === true) {
+          $scope.openModal();
+        }
+      });
+
+      if ($window.innerHeight > 400) {
+        $element.height($window.innerHeight - 70);
+      } else {
+        $element.height(400);
+      }
```

Comment:
400 comes from where? It would be nice if this value were settable. Some wizards could be smaller, some could be larger depending upon the contents of the pages.

This value could be binded to in the html rather than using $element.height()